### ISSUE TITLE
@TagName in a specific situation crashes with "broken LLVM module found: Expected no forward declarations"

### ISSUE THREAD

**Issue by pfgithub** [ziglang/zig#5197]
Zig version 0.6.0+01605a774, Linux x86_64
```zig
const UnionTag = enum { double_pointer, two };
const Union = union(UnionTag) {
    double_pointer: DoublePointer,
    two: void,

    const DoublePointer = struct {
        pub const Tag = UnionTag.double_pointer;
        dbl_ptr: **Union,
    };
};

pub fn main() !void {
    _ = @tagName(Union.DoublePointer.Tag);
}
```
Expected: the program would compile (`zig build-exe main.zig`)
What actually happened:
```
broken LLVM module found: Expected no forward declarations!
!1812 = <temporary!> !DICompositeType(tag: DW_TAG_structure_type, name: "Union", scope: !273, file: !273, line: 2, flags: DIFlagFwdDecl)

This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
```
- Changing `**Union` to `*Union` allows this to compile.
- Removing the second option from `UnionTag` allows this to compile.
- Moving `@tagName` to a declaration inside Union or DoublePointer and referencing that instead allows this to compile.
- `@compileLog`ging `@tagName` prints the expected `"double_pointer"`, and exits before entering code generation.

**Comment by GoogleBot42** [comment]
From discussion in IRC, it seems I encountered this bug as well.  Here's the simplified case.
```zig
const std = @import("std");
const testing = std.testing;
const ArrayList = std.ArrayList;

pub const Expr = union(enum) {
    a: u8,
};

test "example" {
    var abc = ArrayList(*Expr).init(testing.allocator);
}
```

**Comment by czapek1337** [comment]
Currently hitting this when trying to define an `ArrayList` of tagged union pointers, using the workaround from https://github.com/ziglang/zig/issues/5197#issuecomment-623131963 works but it's ugly in this case. Using extern non-tagged unions fail in the same way as the tagged ones.

**Comment by Vexu** [comment]
Fixed in self hosted by 28cbe5e92a36c81177dbcd2f33fc792468c08304
